src/emqx_const_v2.erl

```diff
@@ -0,0 +1,30 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2023 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%
+%% @doc Never update this module, create a v3 instead.
+%%--------------------------------------------------------------------
+
+-module(emqx_const_v2).
+
+-export([ make_tls_root_fun/2
+        ]).
+
+make_tls_root_fun(cacert_from_cacertfile, CADer) ->
+    fun(InputChain) ->
+            case lists:member(CADer, InputChain) of
```

Comment:
I wonder if this will work during an intermediate CA cert renewal.
e.g. client is sending the old cert, but server is already updated with a new one.

Reply:
Added test in https://github.com/emqx/emqx/pull/10553/commits/dce7dfb6160c92fca4509764ff08267ed24ed629

If client use old client cert signed by old intermediate CA, verification pass.
If client send a cert chain with old intermediate CA cert, verification fail.

Reply:
I'm a bit lost.
`CADer` is the certificate (full certificate, including all data) from the trusted pem (the last one in `cacertfile`).
So if the client does not send any CA at all, then this `case` clause will always evaluate to `unknown_ca`.
This means `ssl` lib actually continue to perform the regular chain validation even if `unknown_ca` is returned here?

Reply:
if 'yes', then why would an old CA cause verification to fail ?

Reply:
OTP continue validate with CA cert file and composed with a chain and this fun actually called twice.

Reply:
I have updated the fun always return trusted CA cert that defined in cacertfile.